clients/src/test/java/org/apache/kafka/common/utils/MockTime.java

```diff
@@ -64,7 +64,7 @@ private void maybeSleep(long ms) {
     }
 
     @Override
-    public void sleep(long ms) {
+    public synchronized void sleep(long ms) {
```

Comment:
Can you explain why these changes are needed? The two clock sources are not synchronized in practice, so I thought we shouldn't enforce it in `MockTime` either. But perhaps there's a good reason.